ocaml/typing/solver.ml

```diff
@@ -35,6 +35,7 @@ module Magic_allow_disallow (X : Allow_disallow) :
       type a b l r. (a, b, allowed * r) sided -> (a, b, l * r) sided =
     Obj.magic
 end
+[@@inline]
```

Comment:
This really shouldn't matter. Did you observe it to?

Reply:
This comes with a bunch of other changes so I'm not sure. However, if this functor is not inlined, I imagine calling `disallow_right` would translates to a `caml_apply`?

Reply:
Oh -- perhaps you're right. I was thinking that the functor doesn't actually call the functions in the passed-in module, but that isn't the difference that matters. Never mind.